fili-core/src/main/java/com/yahoo/bard/webservice/metadata/SegmentMetadata.java

```diff
@@ -92,9 +80,13 @@ private SegmentMetadata(Map<String, Set<Interval>> dimensionIntervals, Map<Strin
         return metricIntervals;
     }
 
+    public boolean isEmpty() {
+        return isEmpty;
+    }
+
```

Comment:
Initially, I was thinking it was going to be called possibly many times (each time the data was accessed), but now that I think about it, it will only be called once when the `SegmentMetadataLoader` runs.

I'd be happy to move the calculation into the method call, but if we know it will be called at least once, why not do it in the constructor and more easily cache the value? (yes, it's a trivial calculation, but java lacks easy memoization primitives...)